cpp/include/cudf/detail/iterator.cuh

```diff
@@ -299,6 +300,36 @@ auto inline make_validity_iterator(column_device_view const& column)
   return make_counting_transform_iterator(cudf::size_type{0}, validity_accessor{column});
 }
 
+/**
+ * @brief validity accessor of column with null bitmask
+ * A unary functor returns validity at index `i`.
+ */
+struct validity_accessor_safe {
```

Comment:
> Do we observe divergence because some threads are working on nullable children while others are not?

No divergence but there may be cases where the left column is nullable while right isn't. I could've made the code block https://github.com/rapidsai/cudf/blob/1c3a99d95da4742b332f79b939d99acfed44c219/cpp/include/cudf/table/experimental/row_operators.cuh#L538-L553 significantly more verbose for no practical runtime savings.

Sometimes we just don't need the savings from is_valid_nocheck

Reply:
Made the `validity_accessor` changes